go/oasis-test-runner/scenario/e2e/runtime/runtime_upgrade.go

```diff
@@ -0,0 +1,276 @@
+package runtime
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/oasisprotocol/oasis-core/go/common"
+	"github.com/oasisprotocol/oasis-core/go/common/sgx"
+	keymanager "github.com/oasisprotocol/oasis-core/go/keymanager/api"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/env"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/oasis"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/oasis/cli"
+	"github.com/oasisprotocol/oasis-core/go/oasis-test-runner/scenario"
+	registry "github.com/oasisprotocol/oasis-core/go/registry/api"
+)
+
+// RuntimeUpgrade is the runtime upgrade scenario.
+var RuntimeUpgrade scenario.Scenario = newRuntimeUpgradeImpl()
+
+type runtimeUpgradeImpl struct {
+	runtimeImpl
+
+	nonce uint64
+}
+
+func newRuntimeUpgradeImpl() scenario.Scenario {
+	return &runtimeUpgradeImpl{
+		runtimeImpl: *newRuntimeImpl(
+			"runtime-upgrade",
+			"simple-keyvalue-enc-client",
+			nil,
+		),
+	}
+}
+
+func (sc *runtimeUpgradeImpl) Fixture() (*oasis.NetworkFixture, error) {
+	f, err := sc.runtimeImpl.Fixture()
+	if err != nil {
+		return nil, err
+	}
+
+	// Get number of compute runtimes.
+	var computeIndex int
```

Comment:
Use a negative sentry value and return an error in case there is no compute runtimes? Then you can also remove the duplicate kind check below.